### ISSUE TITLE
Deleted Entity Notification Missing!?

### ISSUE THREAD

**Issue by ptrdtznr** [telefonicaid/fiware-orion#3561]
Lets say a publisher creates an entity, and some subscribers subscribe to it. After some time the publisher deletes the entity. 

Is there any mechanism, that will inform the subscriber about the deleted entity? I couldnt find anything so far about this kind of "feature", "mechansim".

**Comment by fgalan** [comment]
Similar to issue #3560 I think this would be better placed in SOF. Same procedure described here: https://github.com/telefonicaid/fiware-orion/issues/3560#issuecomment-542572102, pls.

Thanks!

**Comment by ptrdtznr** [comment]
I dont think this is an questions, I think this is an issue, since subscribers don't know when the entity has been deleted. And this is a crucial thing which should be somehow handled by Orion. I am not accepting a close and/or a SOF post as an answer.

**Comment by fgalan** [comment]
> I dont think this is an questions

"Is there any mechanism, that will inform the subscriber about the deleted entity?" looks like a question to me ;)

Maybe it's a matter of perspective... Requests like "is there X in Orion?" or "how I can do X in Orion?" are good candidates for SOF. Enrich the Q&A knowledgue base that way is very useful for users. And of course it doesn't preclude that the answer could be "You cannot do X at the present moment" and, in that case, to use a github issue about the feature to implement, linked to the SOF question.

But let's go step by step. First, the question in SOF. Then, if this particular case is not implemented, an issue could be created about it (or reuse the present one) and linked from SOF. Typically, when an issue is associated to an SOF case, when the issue is solved, the SOF questions is edited to provide the solution. That way we have full trazability SOF-github, which is very useful for the users in the FIWARE community.

To fully understand your point of not using SOF... is there any blocking or limitation that precludes you to do that? May I help in some way (e.g. as as your proxy and create the question in your behalf)?

Thanks!

**Comment by ilarimikkonen** [comment]
👍 from our side for the feature. We are developing a component that visualizes data. ATM it's not possible to use subscriptions, as if the entity is deleted / removed, it hangs on the map. we have then resorted to polling and that is not the most efficient way.

**Comment by kzangeli** [comment]
FYI, we're implementing a mechanism to get notifications on deletion of entities for the NGSI-LD API.
GEs that deal with historical data, like Cygnus, QL etc would really need this and I believe it would be a great addition to Orion as well.
Something we should have implemented from the very beginning if you ask me

**Comment by mapedraza** [comment]
I agree with @kzangeli this could be a great addition to Orion as well. As this feature is not semantic dependent, could be a good opportunity to have a contribution to Orion as well. Since this feature is being implemented in Orion-LD by the FIWARE Foundation, there are also plans to contribute it also in this Orion repo?

**Comment by jaimeventura** [comment]
+1 for Orion (NGSI v2),
I'm in a similar situation as @ilarimikkonen.

**Comment by kzangeli** [comment]
https://github.com/FIWARE/context.Orion-LD/issues/280    (Look at the very first line of the ToDo)

**Comment by fgalan** [comment]
I think this is covered by the `alterationTypes` feature introduced in Orion 3.7.0 (see: https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/orion-api.md#subscriptions-based-in-alteration-type). If you use `entityDelete` in the `alterationTypes` array, you will be notified  whenever a entity covered by the subscription is deleted.

@ptrdtznr @ilarimikkonen @jaimeventura could you tell us if this suffices your use case and works for you, please? Thanks!

(Orion 3.7.0 was released time ago... I'm afraid this issue has passed unnoticed until now)

**Comment by ptrdtznr** [comment]
Thats nice that this feature has made it finally :) Unfortunately I am not able to, since my actions using fiware has been faded out...

**Comment by kzangeli** [comment]
FYI (and, sorry for using telefonicas github to disseminate this piece of info about Orion-LD).
I just merged a PR for notifications for `DELETE /ngis-ld/v1/entities/{entityId}`  (forwarded requests as well)
I'm currently working on the same functionality for "BATCH Entity Delete". Hopefully ready some time this week

**Comment by fgalan** [comment]
> Thats nice that this feature has made it finally :) Unfortunately I am not able to, since my actions using fiware has been faded out...

Maybe @ilarimikkonen or @jaimeventura could provide feedback, if they are yet active in FIWARE...
